src/runtime/interop.cs

```diff
@@ -69,27 +68,66 @@ public ModulePropertyAttribute()
         }
     }
 
+    internal static partial class TypeOffset
+    {
+        static TypeOffset()
+        {
+            Type type = typeof(TypeOffset);
+            FieldInfo[] fields = type.GetFields();
+            int size = IntPtr.Size;
+            for (int i = 0; i < fields.Length; i++)
+            {
+                int offset = i * size;
+                FieldInfo fi = fields[i];
+                fi.SetValue(null, offset);
+            }
+        }
+
+        public static int magic() => ManagedDataOffsets.Magic;
```

Comment:
@amos402 actually, this seems to be the line, that contains the actual fix.

Most of other changes are not necessary, and should be excluded from the PR.

I think it is fine to move `TypeOffset` sconstructor from generated files to `interop.cs`, but the template for the `interop*.cs` should also be updated.

Reply:
@lostmsu I figured I'd keep all of the changes so that it'd make the soft shutdown PR smaller and easier to merge.  If you pushback I'll go ahead and trim it down

Reply:
@koubaa let's keep these changes.

There are a few _new_ functions, that are not used anywhere outside of the rest of `soft-shutdown` PR. These should be removed.

Also, we should turn `TypeFlags` into `[Flags]` enum. (at least I do not see any reason not to)